tests/test_barman_cloud_backup_delete.py

```diff
@@ -103,24 +103,73 @@ def _get_mock_backup_files(self, file_paths):
             for name, path in file_paths.items()
         )
 
-    def _create_backup_metadata(
-        self, backup_ids, begin_wals=None, end_wals=None, is_snapshot_backup=False
+    def _create_backup_info(
+        self,
+        backup_id,
+        begin_wals=None,
+        end_wals=None,
+        backup_name=None,
+        parent_backup_id=None,
+        children_backup_ids=None,
+        is_incremental=False,
+        has_children=False,
     ):
         """
-        Helper for tests which creates mock BackupFileInfo and BackupInfo objects
-        which are returned in a dict keyed by backup_id.
+        Helper for tests which creates mock BackupInfo objects.
 
         If begin_wals has an entry for a given backup_id then the begin_wal and
         timeline values will be set according to the wal name in begin_wals.
 
         This is used by tests for two purposes:
-          1. Creating a mock CloudBackupCatalog.
-          2. Providing the data needed to verify that the expected backups were deleted.
+          1. Helper to create a backup_metadata and mock CloudBackupCatalog.
+          2. Providing the data needed to create backup_metadata.
         """
         if begin_wals is None:
             begin_wals = {}
         if end_wals is None:
             end_wals = {}
+        backup_info = mock.MagicMock(name="backup_info", snapshots_info=None)
+        backup_info.backup_id = backup_id
+        if backup_name:
+            backup_info.backup_name = backup_name
+        backup_info.status = "DONE"
+        backup_info.end_time = datetime.datetime.strptime(
+            backup_id, "%Y%m%dT%H%M%S"
+        ) + datetime.timedelta(hours=1)
+        try:
+            backup_info.begin_wal = begin_wals[backup_id]
+            backup_info.timeline = int(backup_info.begin_wal[:8])
+        except KeyError:
+            pass
+        try:
+            backup_info.end_wal = end_wals[backup_id]
+        except KeyError:
+            pass
```

Comment:
```suggestion
        if begin_wal:
            backup_info.begin_wal = begin_wal
            backup_info.timeline = int(backup_info.begin_wal[:8])

        if end_wal:
            backup_info.end_wal = end_wal
```

We would no longer deal with this as a dict.